### PR TITLE
Feature/ticket124

### PR DESCRIPTION
**Pull request by nsahai**
## Description
Added a graceful truncation of Response message once it reached to the HT_DATA_CAPTURE_BODY_MAX_SIZE_BYTES . 

### Testing
Added a new flask test case to set this variable locally and compare truncated message. 
Full set of test cases passed.

### Checklist:
My changes will add something like {"json":["test","test value"],"hypertrace": "truncated"}

### CODE REVIEW

File: src/hypertrace/agent/instrumentation/__init__.py [hunk 1]
```diff
@@ -10,7 +10,18 @@
 # Setup logger name
 logger = logging.getLogger(__name__) # pylint: disable=C0103
 
+
+def is_json(body: str) -> bool:
+    """Checks if body is valid json"""
+    try:
+        json.loads(body)
```
Comment (jcchavezs) [inline src/hypertrace/agent/instrumentation/__init__.py:17]: I am not 100% sure about this. Parsing the JSON body every time seems overkilling, can't we just do a best effort and check if it starts or ends with `[]` or `{}` and add the truncation accordingly.

File: src/hypertrace/agent/instrumentation/__init__.py [hunk 2]
```diff
@@ -33,7 +44,11 @@ def __init__(self):
         self._process_request_body = False
         self._process_response_body = False
         self._service_name = 'hypertrace-python-agent'
-        self._max_body_size = 128 * 1024
+        if 'HT_DATA_CAPTURE_BODY_MAX_SIZE_BYTES' in os.environ \
+                and os.environ["HT_DATA_CAPTURE_BODY_MAX_SIZE_BYTES"]:
+            self._max_body_size = int(os.environ["HT_DATA_CAPTURE_BODY_MAX_SIZE_BYTES"])
+        else:
+            self._max_body_size = 128 * 1024
```
Comment (jcchavezs) [inline src/hypertrace/agent/instrumentation/__init__.py:51]: better move this value to a constant.

File: src/hypertrace/agent/instrumentation/__init__.py [hunk 3]
```diff
@@ -367,6 +382,20 @@ def grab_first_n_bytes(self, body: str) -> str:
...
-            return body[0, self.get_max_body_size()]
+            return self.valid_body(body, self.get_max_body_size())
         else:
             return body
+
+    # Return valid json body
+    def valid_body(self, body: str, index: int) -> str:
+        """Returns the valid Json body"""
+        body = body[0:index]
+        index = index - 1
```
Comment (jcchavezs) [inline src/hypertrace/agent/instrumentation/__init__.py:393]: what if index = 0?

File: src/hypertrace/agent/instrumentation/__init__.py (hunk 2 above)
Comment (nsahai) [reply src/hypertrace/agent/instrumentation/__init__.py:51]: agree

**Pull request closed without merging**